Azure/azure-cli
# [Redis] Add support for `2022-06-01` API version

**Related command**

`az redis`

**Description**

This PR adds supports for a new API version for Azure Cache for Redis (`2022-06-01`). Specifically, these changes include:
- Upgrading `azure-mgmt-redis` package to new version (`14.1.0`), which updates all requests to use the new `2022-06-01` API version
- Add a new optional parameter, `--preferred-data-archive-method` to `az redis import` and `az redis export` operation. 
- Linked server deletion (`az redis server-link delete`) is now a long running operation so it now calls `begin_delete` instead of `delete`. Linked server deletion behavior is unchanged for the user. 
- Updating a cache (`az redis update`) is now a long running operation so it now calls `begin_update` instead of `update`. Implemented a customer setter function so as not to break current behavior. 
- A linked server has two new properties: `geoReplicatedPrimaryHostName` and `primaryHostName`. 
- Relevant updates to help text
- Relevant updates to tests 

**Testing Guide**

Updating a cache (example below shows updating the SKU):
```Python
az redis update --name MyRedisCache --resource-group MyResourceGroup --set "sku.capacity"="2"
```
Linked server operations:
```Python
az redis create -n {primaryName} -g {rg} -l {location} --sku {sku} --vm-size {size} # create primary
az redis create -n {secondaryName} -g {rg} -l {location} --sku {sku} --vm-size {size} # create secondary

az redis server-link create -n {primaryName} -g {rg} --replication-role Secondary --server-to-link {secondaryName} # link servers

az redis server-link show -n {primaryName} -g {rg} --linked-server-name {secondaryName} # see link details

az redis server-link delete -n {primaryName} -g {rg} --linked-server-name {secondaryName} # remove link

```
**History Notes**


[Redis] `az redis import/export`: Add new optional parameter `--preferred-data-archive-method`
[Redis] `az redis server-link`: Linked server has two new properties:  `geoReplicatedPrimaryHostName` and `primaryHostName`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
